modules/statistics/AvgRTPStatsReporter.js

```diff
@@ -695,6 +761,102 @@ export default class AvgRTPStatsReporter {
     }
 
     /**
+     * Calculates average number of pixels for the report
+     *
+     * @param {go figure} resolution
+     * @param {boolean} isLocal if the average is to be calculated for the local
+     * video or <tt>false</tt> if for remote videos.
+     * @param {VideoType} videoType
+     * @return {number|NaN} average number of pixels or <tt>NaN</tt> if there
+     * are no samples.
+     * @private
+     */
+    _calculateAvgVideoPixels(resolution, isLocal, videoType) {
+        let peerPixelsSum = 0;
+        let peerCount = 0;
+        const myID = this._conference.myUserId();
+
+        for (const peerID of Object.keys(resolution)) {
+            if (isLocal ? peerID === myID : peerID !== myID) {
+                const participant
+                    = isLocal
+                    ? null : this._conference.getParticipantById(peerID);
+                const videosResolution = resolution[peerID];
+
+                // Do not continue without participant for non local peerID
+                if ((isLocal || participant) && videosResolution) {
+                    const peerAvgPixels = this._calculatePeerAvgVideoPixels(
+                        videosResolution, participant, videoType);
+
+                    if (!isNaN(peerAvgPixels)) {
+                        peerPixelsSum += peerAvgPixels;
+                        peerCount += 1;
+                    }
+                }
+            }
+        }
+
+        return peerPixelsSum / peerCount;
+    }
+
+    /**
+     * Calculate average pixels for either remote or local participant
+     * @param {object} videos maps resolution per video SSRC
+     * @param {JitsiParticipant|null} participant remote participant or
+     * <tt>null</tt> for local video pixels calculation.
+     * @param {VideoType} videoType the type of the video for which an average
+     * will be calculated.
+     * @return {number|NaN} average video pixels of all participant's videos or
+     * <tt>NaN</tt> if currently not available
+     * @private
+     */
+    _calculatePeerAvgVideoPixels(videos, participant, videoType) {
```

Comment:
if this is used for both local and remote, it seems odd to name it something with "peer" in it (since it handles local, non-peer average as well)